service/tablet_allocator.cc

```diff
@@ -419,6 +422,13 @@ class load_balancer {
 
         utils::chunked_vector<skipped_candidate> skipped_candidates;
 
+        std::optional<double> capacity_per_shard() const {
+            if (!capacity) {
+                return std::nullopt;
+            }
+            return double(*capacity) / shard_count;
```

Comment:
Could write: `return capacity.transform([] (double c) { return c / shard_count; });`. Would be more impressive with less verbose lambdas.

Reply:
Yes, monads :heart_eyes: